doc/kuksa_analysis/kuksa_use_cases.md

```diff
@@ -0,0 +1,243 @@
+# KUKSA Use Cases
+
+
+# Content
+- [KUKSA Use Cases](#kuksa-use-cases)
+- [Content](#content)
+- [Use Cases status](#use-cases-status)
+- [Use cases:](#use-cases)
+  - [UC1](#uc1)
+  - [UC2](#uc2)
+  - [UC3](#uc3)
+  - [UC4](#uc4)
+  - [UC5](#uc5)
+  
+# Use Cases status
+| Description                               | Status                       |
+|-------------------------------------------|------------------------------|
+| Implemented and Verified                  | 🟢                           |
+| Approved, Not Yet Implemented             | 🟡                           |
+| Long Term Goal                            | 🔴                           |
+
+# Use cases:
+
+## UC1
+    Title: Signal consumber subscribes to all signal values.
+
+    Status: 🟡
+
+    Description: Signal Consumer subscribes on changed sensor values or actuator current values of all signals
+
+**Primary Actor:** Signal Consumer
+
+**Secondary Actors:** Databroker, Provider, Vehicle Network 
+
+**Priority**: High
+
+**Preconditions:**
+ * Signal Consumer contains a valid authentication token to connect and perform calls to databroker.
+
+**Special requirements:**
+ * The use case must meet the Performance
+ * What if provider died?
+
+**Assumptions:**
+ * Signal Consumer subscribes to all the possible sensor values or actuator current values to test special requirements.
+ * Provider porvides signals values to Databroker as fast as receives data from the Vehicle Network.
+
+**Postconditions:**
+ * A subscription is created, and Signal Consumer receives streaming of signals.
+
+**Sequence diagram:**
+
+![Signal Consumer Subscribes](../diagrams/Signal_Consumer_subscribes.png)
```

Comment:
My intention for removing some API-specific details is to have the use cases independent of the implementation.
```suggestion
The following sequence shows a situation, where a consumer subscribes to a set of signals supplied by two different providers. One has already connected to the Databroker when the consumer raises its subscription, the other provider is connecting afterwards.

![Signal Consumer Subscribes](../diagrams/Signal_Consumer_subscribes.png)
```